management_api_app/db/repositories/resources.py

```diff
@@ -0,0 +1,26 @@
+from azure.cosmos import CosmosClient
+from pydantic import UUID4
+
+from core import config
+from db.errors import EntityDoesNotExist
+from db.repositories.base import BaseRepository
+from jsonschema import validate
+
+
+class ResourceRepository(BaseRepository):
+    def __init__(self, client: CosmosClient):
+        super().__init__(client, config.STATE_STORE_RESOURCES_CONTAINER)
+
+    @staticmethod
+    def _validate_resource_parameters(workspace_create, workspace_template):
```

Comment:
since this is a generic resource - should we have resource_create and resource_template?